testproject/Assets/Tests/Runtime/NetworkSceneManager/NetworkSceneManagerEventNotifications.cs

```diff
@@ -0,0 +1,265 @@
+            Assert.AreEqual(m_ServerNetworkManager.SceneManager.LoadScene(m_CurrentSceneName, LoadSceneMode.Additive), SceneEventProgressStatus.SceneEventInProgress);
+
+            // Wait for all clients to load the scene
+            yield return WaitForConditionOrTimeOut(ConditionPassed);
+            Assert.IsFalse(s_GlobalTimeoutHelper.TimedOut, $"Timed out waiting for all clients to load {m_CurrentSceneName}");
+
+            // Test unloading additive scenes and the associated event messaging and notification pipelines
+            ResetWait();
+            Assert.AreEqual(m_ServerNetworkManager.SceneManager.UnloadScene(m_CurrentScene), SceneEventProgressStatus.Started);
+
+            yield return WaitForConditionOrTimeOut(ConditionPassed);
+            Assert.IsFalse(s_GlobalTimeoutHelper.TimedOut, $"Timed out waiting for all clients to unload {m_CurrentSceneName}");
+
+            // Check error status for trying to unloading something not loaded
+            ResetWait();
+            Assert.AreEqual(m_ServerNetworkManager.SceneManager.UnloadScene(m_CurrentScene), SceneEventProgressStatus.SceneNotLoaded);
+
+            // Check error status for trying to load an invalid scene name
+            LogAssert.Expect(LogType.Error, $"Scene '{k_InvalidSceneName}' couldn't be loaded because it has not been added to the build settings scenes in build list.");
+            Assert.AreEqual(m_ServerNetworkManager.SceneManager.LoadScene(k_InvalidSceneName, LoadSceneMode.Additive), SceneEventProgressStatus.InvalidSceneName);
+
+            ShutdownAndCleanUp();
+        }
+
+        /// <summary>
+        /// Resets each SceneTestInfo entry
+        /// </summary>
+        private void ResetWait()
+        {
+            foreach (var entry in m_ShouldWaitList)
+            {
+                entry.ShouldWait = true;
+                entry.ProcessedEvent = false;
+            }
+        }
+
+        /// <summary>
+        /// Initializes the m_ShouldWaitList
+        /// </summary>
+        private void InitializeSceneTestInfo(LoadSceneMode clientSynchronizationMode, bool enableSceneVerification = false)
+        {
+            m_ShouldWaitList.Add(new SceneTestInfo() { ClientId = m_ServerNetworkManager.ServerClientId, ShouldWait = false });
+
+            foreach (var manager in m_ClientNetworkManagers)
+            {
+                m_ShouldWaitList.Add(new SceneTestInfo() { ClientId = manager.LocalClientId, ShouldWait = false });
+            }
+        }
+
+        /// <summary>
+        /// Wait until all clients have processed the event and the server has determined the event is completed
+        /// Will bail if it takes too long via m_TimeOutMarker
+        /// </summary>
+        private bool ConditionPassed()
+        {
+            return !(m_ShouldWaitList.Select(c => c).Where(c => c.ProcessedEvent != true && c.ShouldWait == true).Count() > 0);
```

Comment:
`!(... > 0)` vs `(... == 0)`? :P

Reply:
Oh... hahah... I don't mind changing that... they pretty much are the same cost in assembly instructions as well as the same logical translation...but I can see where it might seem clearer to be == 0.